app/scripts/app.js

```diff
@@ -17,7 +23,8 @@ blocChat.config(['$stateProvider', '$locationProvider', function($stateProvider,
 
 blocChat.factory("chatRooms", [
     "$firebaseArray",
-    function($firebaseArray) {
+    "CurrentRoom",
+    function($firebaseArray, CurrentRoom) {
         var roomsUrl = 'https://stewart-bloc-chat.firebaseio.com/rooms';
         var roomsRef = new Firebase(roomsUrl);
         var roomsObject = $firebaseArray(roomsRef);
```

Comment:
I'm not sure exactly which line you mean.  I think we'll have to talk through this one.  Sorry man.